### PR TITLE
Blackify code base

### PR DESCRIPTION
**Pull request by Cnidarias**
Letting black run over everything

- [ ] Add documentation.
- [ ] Add a changelog entry.
- [x] Add your name in the contributors file.

So in light of my other pull request I suggested that we run black over everything and @leplatrem seemed to be in favor so here we are.

I figured it would be best if we did this before adding all the type annotations to not have too many changes at once.

I let black run over everything, adding a pre-commit (https://pre-commit.com) config file as well as a config file for black.

I also had to add an ignore E203 in order for flake8 to be happy.

If this is okay then I would still have to add some documentation on how to set things up for development but I do think the result is actually quite nice!

Thoughts?

### CODE REVIEW

**Comment by leplatrem** [comment]
@Natim ?

**Comment by leplatrem** [comment]
Note: @peterbe recommends therapist https://github.com/peterbe/hashin/pull/80/files

**Comment by leplatrem** [comment]
Be careful commit edbf3b8  should not be part of this PR

**Comment by Cnidarias** [comment]
So I basically just added the stuff you linked -- not sure I did it correctly though especially regarding therapist/travis/tox

But it seems like its fine?

Still need to add some documentation though.

**Comment by Cnidarias** [comment]
Hey,

yes the diff is quite massive. 
This does beg another question, when looking at the diffs alot of changes are changing single quotes to double quotes

Black does provide the option to omit normalizing quotes. 

After testing it - the diff would still be massive. When omitting the quite normalization black still touches all but 24 files. 


I have also added some documentation now. 
I would prefer to rebase once we are happy with the documentation and everything else because there will always be merge conflicts as of now.

File: docs/community.rst [hunk 1]
```diff
@@ -139,17 +139,27 @@ Get started!
 
    Now you can make your changes locally.
 
-6. When you're done making changes, check that your changes pass the tests::
+6. When you're done making changes, format your code with `black <https://github.com/ambv/black>`_ (black requires python >= 3.6)::
+
+    python -m pip install black
+    black .
```
Comment (leplatrem) [inline docs/community.rst]: I would suggest to install `black` only and run it via `tox -e lint`  (tox and therapist will be installed via dev-requirements)

File: docs/community.rst [hunk 2]
```diff
@@ -139,17 +139,27 @@ Get started!
...
+
+7. Don't forget to check that your changes pass the tests::
 
     make tests
 
-7. Commit your changes and push your branch to GitHub::
+8. (Optional) Install a git hook with `therapist <https://github.com/rehandalal/therapist>`_ ::
+
+    python -m pip install therapist
```
Comment (leplatrem) [inline docs/community.rst]: this line can be omitted since already in dev-requirements

**Comment by leplatrem** [comment]
Almost there! :tada: 

* I would like to avoid configuration of black if possible. Double quotes are fine.
* Since python 3.5 is still officially supported, we have to adapt. Suggesting its installation manually in the dev docs is fine IMO

Thanks :pray:

**Comment by leplatrem** [comment]
Let me know if you need help or support ;) 

Actually I would be very interested with this PR, because I started to work on #710 using `xargs sed` like an animal :)

**Comment by Cnidarias** [comment]
Okay - so I went ahead and got rid of my changes because I did not feel like merging 40+ merge conflicts.

This commit now has everything we had earlier with the additional changes that we now have a lint-requirements.txt file as well as adapting the documentation.

The code has not yet been formatted with black so it will be easier for me to rebase it once we are ready to merge.

File: lint-requirements.txt [hunk 3]
```diff
@@ -0,0 +1,2 @@
+therapist==1.5.0
+black==18.9b0
```
Comment (leplatrem) [inline lint-requirements.txt]: Maybe we could add:
```
    flake8
    setuptools<36
```

BTW @Natim do you remember why `setuptools<36`?

File: docs/community.rst [hunk 4]
```diff
@@ -139,17 +139,26 @@ Get started!
...
+
+7. Don't forget to check that your changes pass the tests::
 
     make tests
 
-7. Commit your changes and push your branch to GitHub::
+8. (Optional) Install a git hook::
+
+    therapist install
+
```
Comment (leplatrem) [inline docs/community.rst]: looks like an extra line

File: tox.ini [hunk 5]
```diff
@@ -16,11 +16,12 @@ deps =
     statsd
 install_command = pip install {opts} {packages}
 
-[testenv:flake8]
-commands = flake8 kinto tests docs/conf.py
```
Comment (leplatrem) [inline tox.ini]: shouldn't you keep `flake8` for py35 on Travis?

File: docs/community.rst (hunk 4 above)
Comment (Cnidarias) [reply docs/community.rst]: Don't you have to tell therapist to create the hook in your .git dir?

File: tox.ini (hunk 5 above)
Comment (Cnidarias) [reply tox.ini]: Yes I should have removed it in travis.

**Comment by leplatrem** [comment]
:) here we go :) thanks again ;)

**Pull request merged**